packages/smart-contracts/src/contracts/EthereumFeeProxy.sol

```diff
@@ -1,78 +1,75 @@
+     * @param _paymentReference Reference of the payment related
+     * @param _feeAmount The amount of the payment fee (part of the msg.value)
+     * @param _feeAddress The fee recipient
+     */
+    function transferWithReferenceAndFee(
+        address payable _to,
+        bytes calldata _paymentReference,
+        uint256 _feeAmount,
+        address payable _feeAddress
+    ) external payable {
+        transferExactEthWithReferenceAndFee(
+            _to,
+            msg.value - _feeAmount,
+            _paymentReference,
+            _feeAmount,
+            _feeAddress
+        );
+    }
 
-  /**
-  * @notice Performs an Ethereum transfer with a reference
-  * @param _to Transfer recipient
-  * @param _paymentReference Reference of the payment related
-  * @param _feeAmount The amount of the payment fee (part of the msg.value)
-  * @param _feeAddress The fee recipient
-  */
-  function transferWithReferenceAndFee(
-    address payable _to,
-    bytes calldata _paymentReference,
-    uint256 _feeAmount,
-    address payable _feeAddress
-  )
-    external
-    payable
-  {
-    transferExactEthWithReferenceAndFee(
-      _to,
-      msg.value - _feeAmount,
-      _paymentReference,
-      _feeAmount,
-      _feeAddress
-    );
-  }
+    /**
+     * @notice Performs an Ethereum transfer with a reference with an exact amount of eth
+     * @param _to Transfer recipient
+     * @param _amount Amount to transfer
+     * @param _paymentReference Reference of the payment related
+     * @param _feeAmount The amount of the payment fee (part of the msg.value)
+     * @param _feeAddress The fee recipient
+     */
+    function transferExactEthWithReferenceAndFee(
+        address payable _to,
+        uint256 _amount,
+        bytes calldata _paymentReference,
+        uint256 _feeAmount,
+        address payable _feeAddress
+    ) public payable nonReentrant {
+        (bool sendSuccess, ) = _to.call{value: _amount}('');
+        require(sendSuccess, 'Could not pay the recipient');
 
+        _feeAddress.transfer(_feeAmount);
 
-  /**
-  * @notice Performs an Ethereum transfer with a reference with an exact amount of eth
-  * @param _to Transfer recipient
-  * @param _amount Amount to transfer
-  * @param _paymentReference Reference of the payment related
-  * @param _feeAmount The amount of the payment fee (part of the msg.value)
-  * @param _feeAddress The fee recipient
-  */
-  function transferExactEthWithReferenceAndFee(
-    address payable _to,
-    uint256 _amount,
-    bytes calldata _paymentReference,
-    uint256 _feeAmount,
-    address payable _feeAddress
-  )
-    nonReentrant
-    public
-    payable
-  {
-    _to.transfer(_amount);
-    _feeAddress.transfer(_feeAmount);
-    // transfer the remaining ethers to the sender
-    payable(msg.sender).transfer(msg.value - _amount - _feeAmount);
+        // transfer the remaining ethers to the sender
+        (bool sendBackSuccess, ) = payable(msg.sender).call{
```

Comment:
Modified `.transfer` to `.call` here too. This way we cover the case where a gnosis safe is the one initiating a payment and we have to send ether back to it.